sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -136,6 +139,26 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
     }
     SparkSession.sqlListener.get()
   }
+
+  /*
+   * This belongs here more than in `SessionState`. However, does not seem that it can be
```

Comment:
also i think its more obvious here (since it uses jarClassloaded from sharedState). so no reasoning necessary at all.